plugins/gatsby-theme-iterative-docs/src/utils/front/Slugger.ts

```diff
@@ -0,0 +1,43 @@
+class Slugger {
+  separator: string
+  lowercase: boolean
+  slugs: Array<string>
+
+  constructor(options?: { separator?: string; lowercase?: boolean }) {
+    this.separator = options?.separator || '-'
+    this.lowercase = options?.lowercase === false ? false : true
+    this.slugs = []
+  }
+
+  slug(str: string) {
+    str = typeof str === 'string' ? str : ''
+    let slug = this.slugify(str)
+
+    if (this.lowercase) {
+      slug = slug.toLowerCase()
+    }
+    if (this.slugs.includes(slug)) {
+      throw new Error(`Duplicate slug: ${slug}`)
+    }
+    this.slugs.push(slug)
+    return slug
+  }
+  slugify = (str: string) => {
+    return str
+      .replace(/[^\w\s-]/g, '')
+      .trim()
+      .replace(/[-\s]+/g, this.separator)
+      .replace(this.separator + this.separator, this.separator)
+      .replace(/(^\-+|\-+$)/g, '')
+  }
+  add(slug: string) {
```

Comment:
Should we use the `add` function inside of the `slug` function? That way the `throw new Error` code is only in one spot?

Reply:
Actually, ended up not using it so can drop this function.